examples/voice_agents/awaitable_interrupt.py

```diff
@@ -0,0 +1,66 @@
+import logging
+
+from dotenv import load_dotenv
+
+from livekit import rtc
+from livekit.agents import (
+    Agent,
+    AgentSession,
+    JobContext,
+    WorkerOptions,
+    cli,
+)
+from livekit.plugins import openai, silero
+
+logger = logging.getLogger("interrupt-example")
+logger.setLevel(logging.INFO)
+
+load_dotenv()
+
+
+# this example demonstrates how to manually interrupt voice agent.
+# session.interrupt() provides a safe way to cancel the current generation. this method
+# returns a Future that must be awaited to ensure the chatcontext is fully updated before
+# accessing it.
+
+
+class Agent(Agent):
+    def __init__(self) -> None:
+        super().__init__(
+            instructions="You are a helpful voice assistant",
+        )
+
+
+async def entrypoint(ctx: JobContext):
+    await ctx.connect()
+
+    session = AgentSession(
+        vad=silero.VAD.load(),
+        stt=openai.STT(),
+        llm=openai.LLM(model="gpt-4o-mini"),
+        tts=openai.TTS(voice="alloy"),
+    )
+
+    @ctx.room.local_participant.register_rpc_method("emergency_interrupt")
+    async def on_emergency_interrupt(data: rtc.RpcInvocationData) -> None:
```

Comment:
It's kind of misleading to use a function tool to interrupt? Taking this example from a user perspective, it's hard to understand what is on_emergency_interrupt really doing.

`await session.interrupt()` isn't very useful inside a function_call